docs/user_guide/configuration.rst

```diff
@@ -0,0 +1,228 @@
+                  "stream": "ext://sys.stdout"
+              }
+          },
+          "loggers":{
+              "idaes":{
+                  "level": "INFO",
+                  "propagate": true,
+                  "handlers": ["console"]
+              },
+              "idaes.solve":{
+                  "propagate": false,
+                  "level": "INFO",
+                  "handlers": ["console"]
+              },
+              "idaes.init":{
+                  "propagate": false,
+                  "level": "INFO",
+                  "handlers": ["console"]
+              },
+              "idaes.model":{
+                  "propagate":false,
+                  "level": "INFO",
+                  "handlers": ["console"]
+              }
+          }
+      }
+  }
+
+Global Configuration Files
+--------------------------
+
+IDAES configuration files are named ``idaes.conf``. The easiest way to find
+where the global configuration file should be placed is to run the command
+``idaes data-directory``.  A global configuration file won't exist unless a
+user creates one. The default configuration above can be used as a start.
+
+Windows
+~~~~~~~
+
+On Windows the global configuration file is located at
+``%LOCALAPPDATA%\idaes\idaes.conf``.
+
+UNIX-Like
+~~~~~~~~~
+
+On Unix-like systems the global configuration files is located at
+``$HOME/.idaes/idaes.conf``.
+
+Other
+~~~~~
+
+On systems that have neither an ``%LOCALAPPDATA%`` or ``$HOME`` environment
+variable, global config files are not currently supported.
+
+Local Configuration Files
+-------------------------
+
+Local configuration files are also named ``idaes.conf`` and can be placed in the
+working directory, which is the directory you launch Python from.  You can also
+use the Python command ``chdir()`` to change the working directory before
+importing ``idaes``.
+
+In addition to reading local configuration files when ``idaes`` is imported, you
+can read a configuration file anytime by calling ``idaes.read_config(path)``.
+Reading a configuration file will automatically apply any resulting
+configuration changes.
+
+Changing the Configuration in a Script or Interactive Session
+-------------------------------------------------------------
+
+The idaes configuration can be changed anytime after the ``idaes`` module is
+imported.  The standard ConfigBlock options are described in detail below.  For
+example to change whether you want to use the solvers provided by idaes or ones
+you have installed elsewhere, you would first use the command
+``idaes.cfg["use_idaes_solvers"] = False`` then to make the change take effect
+use ``idaes.reconfig()``.  Not all option changes require ``idaes.reconfig()``,
+so whether they do or don't is provided in the options descriptions below.
+
+Important Configuration Entries
+-------------------------------
+
+The ConfigBlock has several options, but they are not all important to
+end-users. This section lists the commonly used entries.
+
```

Comment:
This shouldn't hold up this PR, but one of the driving motivations behind the Config system was to include documentation *in* the configuration.  So, we *could* move this into the Config definitions and then autogenerate this text here.

Reply:
Yeah.  That's a good point, and I should have thought of that because we document our models that way.  I'll go ahead and switch this over if it's not too hard.